driver-core/src/main/com/mongodb/client/model/changestream/UpdateDescription.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.changestream;
+
+import org.bson.Document;
+import org.bson.codecs.pojo.annotations.BsonCreator;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+import java.util.List;
+
+/**
+ * The update description for changed fields in a {@code $changeStream} operation.
+ *
+ * @since 3.6
+ */
+public final class UpdateDescription {
+    private final List<String> removedFields;
+    private final Document updatedFields;
```

Comment:
Are there any other models that use the Document class?  I was expecting BsonDocument instead.

Reply:
While BsonDocument is more awkward to use than Document, it's the safer choice here, as we don't know how the Document codec is configured.

Looking at the rest of the driver,  there's not that many places we've had to make this choice, but in most of them we've gone with BsonDocument/BsonValue, including:

* MongoCommandException
* BulkWriteUpsert
* UpdateResult

With BsonDocument you know what you've got, and can always go from that to a Document with no loss of information.

Reply:
Done